### PR TITLE
Add new game prompt

### PR DESCRIPTION
**Pull request by Struckdown**
When the game ends, a new game of TicTacToe can be started from within the program, maintaining the score

### CODE REVIEW

**Comment by MichaelParadis** [comment]
@ZhimaoLin @pshn111 please tag me when you all are done reviewing this so I can take a look at it. Thanks

File: src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java [hunk 1]
```diff
@@ -45,12 +55,38 @@ public void playGame(){
...
+        printScore();
+    }
+
+    public boolean promptNewGame() {
+        Scanner keyboardScanner = new Scanner(System.in);
+        System.out.println("Do you want to play another game (y/n)?");
+        String line = keyboardScanner.nextLine().toLowerCase();
+        if (line.contentEquals("y")) {
+            return true;
+        } else {
```
Comment (pshn111) [inline src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java]: I think you should add a User Input Error Handling.
If a user wants to input "y", but user inputs some other character by mistake. In this case, I think you should ask the user again.

File: src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java [hunk 2]
```diff
@@ -29,6 +34,11 @@ public void promptNextPlayer(){
         }
     }
 
+    public void printScore(){
+        System.out.println("X wins: " + score[0]  + "    O wins: " + score[1] +
+                    "    Ties:" + score[2]+ "");
+    }
```
Comment (pshn111) [inline src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java:42]: The feature 2 requirement shows to keep track of how many games were won, lost, tied by each player.
Therefore, you should add a lost score.

File: src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java [hunk 3]
```diff
@@ -12,11 +12,16 @@
 public class TicTacToeGame {
 
     private Board board;
+    private int[] score = {0,0,0};	// X wins, O wins, Ties
```
Comment (pshn111) [inline src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java]: It is better to initial score in resetBoard.

**Review by pshn111 — DISMISSED** [review]
You need to fix problems for the scoreboard.
It is better to have a User Input Error Handling.

**Review by pshn111 — CHANGES_REQUESTED** [review]
You need to fix problems for the scoreboard (pull request #3 ).
It is better to have a User Input Error Handling.
You need to fix pull request #3 first.

**Review by ZhimaoLin — CHANGES_REQUESTED** [review]
Please also print the loss score of each player.

File: src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java (hunk 3 above)
Comment (MichaelParadis) [reply src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java]: reset board clears the game moves if this is set there the score will not be tracked. I moved this to the constructor of TicTacToeGame instead.

**Comment by MichaelParadis** [comment]
@pshn111 @ZhimaoLin updated to address feedback

**Pull request merged**